### ISSUE TITLE
Implement API to fetch internships

### ISSUE THREAD

**Issue by jwu910** [BobaTalks/internship-tracker#108]
## Description

Blocked by #35

Once the above ticket is completed, we will need to implement the API method that will make the requests to the provided endpoint.

Optionally we can consider React Router's Loader method as a pattern to observe

https://reactrouter.com/en/main/route/loader

React router's loader feature allows initiating fetch requests to happen sooner to help mitigate double renders and ideally the promise of data sooner

**Comment by a-camarillo** [comment]
@jwu910 what are the thoughts with this? Do we want to implement the react router loader method or opt for using something like fetch/axios for making requests.

I personally think using fetch and axios will be faster to build since I'm familiar with those but the loader can offer some benefits I've just never used it

**Comment by jwu910** [comment]
We can skip the React router implementation. It's more of a scalability and nice to have feature. We can just use (preferably) axios for the api calls
